### PR TITLE
[assessment-tool] Impemented deletion of non-found samples.

### PR DESCRIPTION
**Pull request by ygorshenin**
@mpimenov @milchakov PTAL

Deletion is implemented in a lazy way - deleted items are actually marked as "deleted", and won't be saved or displayed to the user.

Pros of this approach:
* no dangling references/pointers, no deletions in vectors
* instead of changing visibility of "deleted" items we can, for example, set red background for such items, so this approach allows us to experiment with UX

Cons of this approach:
* need to handle "deleted" state for all samples in the model and in the view

### CODE REVIEW

File: search/search_quality/assessment_tool/edits.cpp [hunk 1]
```diff
@@ -15,52 +15,89 @@ bool Edits::RelevanceEditor::Set(Relevance relevance)
...
 // Edits -------------------------------------------------------------------------------------------
-void Edits::ResetRelevances(std::vector<Relevance> const & relevances)
+void Edits::Apply()
 {
-  WithObserver(Update::AllRelevancesUpdate(), [this, &relevances]() {
-    m_origRelevances = relevances;
-    m_currRelevances = relevances;
+  WithObserver(Update::MakeAll(), [this]() {
+    for (auto & entry : m_entries)
+      entry.m_curr = entry.m_orig;
```
Comment (milchakov) [inline search/search_quality/assessment_tool/edits.cpp]: не понял, текущему присваивается оригинальный, выглядит будто это метод reset а не apply. Проясни пожалуйста

Comment (ygorshenin) [inline search/search_quality/assessment_tool/edits.cpp]: Nice catch. This is an error, the correct version is:
```c++
entry.m_orig = entry.m_curr;
```

**Comment by ygorshenin** [comment]
@milchakov @mpimenov PTAL

File: search/search_quality/assessment_tool/main_view.hpp [hunk 2]
```diff
@@ -32,16 +32,24 @@ class MainView : public QMainWindow, public View
   void OnSearchCompleted() override;
   void ShowSample(size_t sampleIndex, search::Sample const & sample, bool positionAvailable,
                   bool hasEdits) override;
+
   void ShowFoundResults(search::Results::ConstIter begin, search::Results::ConstIter end) override;
-  void ShowNonFoundResults(std::vector<search::Sample::Result> const & results) override;
+  void ShowNonFoundResults(std::vector<search::Sample::Result> const & results,
+                           std::vector<Edits::Entry> const & entries) override;
+
+  void ShowFoundResultsMarks(search::Results::ConstIter begin,
```
Comment (mpimenov) [inline search/search_quality/assessment_tool/main_view.hpp]: Может, передавать сюда тоже вектор, как в ShowNonFoundResultsMarks?
Или наоборот, туда передавать итераторы?

File: search/search_quality/assessment_tool/edits.hpp [hunk 3]
```diff
@@ -13,32 +13,42 @@
 class Edits
 {
 public:
+  using Relevance = search::Sample::Result::Relevance;
+
+  struct Entry
+  {
+    Entry() = default;
+
+    Relevance m_curr = Relevance::Irrelevant;
```
Comment (milchakov) [inline search/search_quality/assessment_tool/edits.hpp]: а мне наоборот больше новые названия нравятся, коротко и понятно

Comment (ygorshenin) [inline search/search_quality/assessment_tool/edits.hpp]: +1, from my point of view datatype + name suggests the correct meaning of the variable.

File: search/search_quality/assessment_tool/edits.hpp [hunk 4]
```diff
@@ -13,32 +13,42 @@
...
+    Relevance m_orig = Relevance::Irrelevant;
```
Comment (ygorshenin) [inline search/search_quality/assessment_tool/edits.hpp]: I'd prefer the current naming.

File: search/search_quality/assessment_tool/main_view.hpp (hunk 2 above)
Comment (ygorshenin) [inline search/search_quality/assessment_tool/main_view.hpp]: Why? It's easy to implement but not needed now.

**Comment by ygorshenin** [comment]
How about to merge this?

**Pull request merged**